### PR TITLE
34 implement storing suspicious deals

### LINKED ISSUE #34: Implement storing suspicious deals
Permanently store suspicious deals

### PR DESCRIPTION
**Pull request by JakubDuch7790**
Storing incoming suspicious deals secured by class SuspiciousTradesInserter via secondary DbContext SuspiciousTradesDbContext.

### CODE REVIEW

File: src/WatchDogs.Application/Program.cs [hunk 1]
```diff
@@ -97,7 +109,9 @@
 
         var loadedTrades = await SDD.LoadDealsAsync();
 
-        await SDD.DetectSuspiciousDealsAsync(loadedTrades);
+        var SS = await SDD.DetectSuspiciousDealsAsync(loadedTrades);
```
Comment (martinbalucha) [inline src/WatchDogs.Application/Program.cs:112]: Why is the using scope still there? I thought we would move it all to `FakeWatcher`

File: src/WatchDogs.Contracts/IDataInserter.cs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+﻿namespace WatchDogs.Contracts;
+public interface IDataInserter
```
Comment (martinbalucha) [inline src/WatchDogs.Contracts/IDataInserter.cs]: The interface is named `IDataInserter` but you are only inserting Trades.

File: src/WatchDogs.Contracts/IDataInserter.cs [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+{
+    Task InsertTradeDatatoDbAsync(IEnumerable<Trade> data);
```
Comment (martinbalucha) [inline src/WatchDogs.Contracts/IDataInserter.cs]: Why not simply `InsertAsync`?

Comment (JakubDuch7790) [reply src/WatchDogs.Contracts/IDataInserter.cs]: I can rename it. Done.

File: src/WatchDogs.Contracts/IDataInserter.cs (hunk 2 above)
Comment (JakubDuch7790) [reply src/WatchDogs.Contracts/IDataInserter.cs]: How would you name it if there already is ITradeInserter? I could rename IDataInserter to ITradeInserter but what would be good name for old  ITradeInserter? IUnfilteredTradesInserter? INormalTradesInserter? Like this it at least make sense for me.

File: src/WatchDogs.Application/Program.cs (hunk 1 above)
Comment (JakubDuch7790) [reply src/WatchDogs.Application/Program.cs:112]: testing purposes

File: src/WatchDogs.Contracts/IDataInserter.cs (hunk 2 above)
Comment (martinbalucha) [reply src/WatchDogs.Contracts/IDataInserter.cs]: I am a bit confused here. We have `ISuspiciousDealInserter`, `ITradeInserter `and `IDataInserter`. I can assume the purpose of the first one but what should be the distinction between the latter two?

Comment (JakubDuch7790) [reply src/WatchDogs.Contracts/IDataInserter.cs]: aha, by using rubber duck method I've just found out that IDataInserter is redundant and ISuspiciousDealInserter can implement ITradeInserter directly

**Pull request merged**